### DISCUSSION TITLE
Interative Dashboard

### DISCUSSION THREAD

**Discussion by joejnt** [teamalgoritma/community#606] [Q&A] [answered]
Dear rekan2 algoritma,

Saya mendapatkan permasalahan untuk menampilkan plot di dashboard dimana syntax sbb:
data yang saya gunakan adalah:

**dataset**
curr_bal <- read.xlsx("Data Input/Kas Khasanah Current Balance.xlsx", sheet = "Query1")
current <- curr_bal %>% 
  mutate(balance = current_bal/1000000) %>% 
  select(c(initial, balance))
empty_bar <- 10
to_add <- matrix(NA, empty_bar, ncol(current))
colnames(to_add) <- colnames(current)
data_currbal <- rbind(current, to_add)
data_currbal$id <- seq(1, nrow(data_currbal))
label_data <- data_currbal
number_of_bar <- nrow(label_data)
angle <- 90 - 360 * (label_data$id-0.5) / number_of_bar
label_data$hjust <- ifelse( angle < -90, 1, 0)
label_data$angle <- ifelse(angle < -90, angle+180, angle)

**server.R**
server <- function(input, output){
  
    output$plot_reg <- renderPlotly({
      plot_reg <- ggplot(data = label_data, aes(x = as.factor(id), y = balance)) +
        geom_bar(stat = "identity", fill=alpha("forestgreen", 0.3)) +
        ylim(-8100, 9000) +
        #  ylim((max(data_currbal$balance)-100)*-1, max(data_currbal$balance)) +
        theme_minimal() +
        theme(
          axis.text = element_blank(),
          axis.title = element_blank(),
          panel.grid = element_blank(),
          plot.margin = unit(rep(-3,6), "cm")
        ) +
        coord_polar(start = 0) + 
        geom_text(data=label_data, 
                  aes(x=id, y=balance+10, 
                      label=initial, 
                      hjust=hjust), color="black", 
                  fontface="bold",
                  alpha=1, 
                  size=2.5, 
                  angle= label_data$angle, 
                  inherit.aes = FALSE)
      plot_reg
    })
  
}

**ui.R**
ui <- dashboardPage(
  dashboardHeader(title = "Surveillance Dept", titleWidth = 230),
  dashboardSidebar(
    sidebarMenu(
      menuItem(text = "Cash Vault", tabName = "cash_vault", icon = icon("vault")),
      menuItem(text = "Funding (CASATD)", tabName = "casatd", icon = icon("building-columns")),
      menuItem(text = "Loan", tabName = "loan", icon = icon("landmark-dome")),
      menuItem(text = "Transaction", tabName = "loan", icon = icon("handshake"))
      
    )
    
    
    
  ),
  dashboardBody(
    tabItems(
      tabItem(tabName = "cash_vault",
              box(
                  title = "Saldo Kas Khasanah Cabang & Capem per Regional",
                  width = 12,
                  plotlyOutput(outputId = "plot_reg")
              )
        
      )
      
    )
    
  ),
  skin = "green" 
)

Namun hasil yang saya dapatkan _error attempt to apply non-function_
hasil capture
<img width="560" alt="image" src="https://user-images.githubusercontent.com/103427989/229289373-b9b5f5f8-7077-4c56-b439-5bbe0259d4f4.png">

mohon sarannya.

Best regards,
JT

Note: untuk plot saya dapatkan referensi dari data-to-viz

**Comment by Yusraf002** [comment]
Dear @joejnt ,
Terima kasih atas pertanyaannya pak, tim kami akan segera coba bantu atau carikan referensi ya pak

**Comment by kevbow** [comment]
Dear @joejnt,

Penggunaan `renderPlotly()` dan `plotlyOutput()` khusus untuk visualisasi yang merupakan plotly object (atau yang disebut sebagai plot interaktif ketika kelas). Sedangkan plot yang anda buat menggunakan ggplot biasa sehingga seharusnya bukan menggunakan kedua fungsi di atas tetapi menggunakan `renderPlot()` dan `plotOutput()`.
Demikian yang dapat saya sampaikan, semoga membantu.

**Comment by joejnt** [comment, marked as answer]
Dear @kevbow 

Well noted, terima kasih untuk inside nya.
Satu lagi, klo mau bkin plot dengan leaflet ada referensi ngga ya untuk donlot daftar long lat nya yang sudah dalam bentuk RDS.

Best regards,
JT

**Reply by kevbow** [reply]
Dear @joejnt,

Perlu informasi spesifik terkait lokasi yang ingin digunakan, lalu akurasinya apakah negara, provinsi, kota, dsb.
Jika informasi umum seperti negara, provinsi, kota biasanya sudah ada yang menyediakan tetapi hal ini belum bisa saya pastikan apakah terdapat dalam bentuk data tabular atau tidak.

**Reply by joejnt** [reply]
DEar @kevbow 

untuk plotnya hanya sebatas peta indonesia, tidak perlu sampai nama jalan tapi hanya sampai pada level kota saja

Best regards,
JT

**Reply by kevbow** [reply]
Dear @joejnt,

Informasi ini banyak tersedia jika Anda melakukan searching google, terutama jika lokasi yang diangkat merupakan area yang umum seperti kota, kabupaten, provinsi, dsb. Berikut contohnya:
- https://simplemaps.com/data/id-cities
- https://datahub.io/JemputData/location_id

Demikian dan terima kasih.

**Reply by joejnt** [reply]
Dear @kevbow 

Well noted, terima kasih untuk informasinya.

Best regards,
Jt
